src/libraries/System.Private.Uri/src/System/ValueStringBuilderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Diagnostics;
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void AppendString(string value) => AppendSlow(value);
```

Comment:
Why can't you use regular `Append(string)`?

Reply:
The idea was to avoid inlining the rest of the `Append(string?)` logic in cases where `AppendSlow` would end up being called every time (like when appending the Uri scheme that is never null or just 1 char).
```cs
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public void Append(string? s)
{
	if (s == null)
	{
		return;
	}

	int pos = _pos;
	if (s.Length == 1 && (uint)pos < (uint)_chars.Length)
	{
		_chars[pos] = s[0];
		_pos = pos + 1;
	}
	else
	{
		AppendSlow(s);
	}
}
```

Reply:
It's possible VSB.Append's implementation should be changed, such that call sites for which it's currently optimizing (e.g. one's where a single-char string is expected) do the work themselves to switch to using Append(char).